### PR TITLE
Add recipe for traittypes

### PR DESCRIPTION
**Pull request by SylvainCorlay**

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
Things seem to be failing on Python 3 only.

File: recipes/traittypes/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    md5: {{ md5 }}
+
+build:
+    number: 0
+    script: python setup.py install
+
+requirements:
+    build:
+        - python
+    run:
```
Comment (jakirkham) [inline recipes/traittypes/meta.yaml]: Missing `- python`.

**Comment by jakirkham** [comment]
So Python has to be listed as a `build` and a `run` dependency. Otherwise `conda` will not recognize it as a Python package. This is almost certainly why the build is failing. Also it is why the Python version is not getting added to the build string.

**Comment by jakirkham** [comment]
Also feel free to merge/rebase with `master` and clear out those other recipes that are not from this PR.

File: recipes/traittypes/meta.yaml (hunk 1 above)
Comment (SylvainCorlay) [inline recipes/traittypes/meta.yaml]: fixed, but still not building!

Comment (jakirkham) [inline recipes/traittypes/meta.yaml]: Are you meaning Travis CI? If so, they are having some infrastructure issues. It seems CircleCI built ok. Not sure what is up with AppVeyor yet.

**Comment by jakirkham** [comment]
The AppVeyor failure is not due to this recipe. Please merge with `master` here at conda-forge and delete all recipes not proposed in this PR (except the `example`).

**Comment by SylvainCorlay** [comment]
Woot! All checks have passed!

**Comment by SylvainCorlay** [comment]
Merge?

**Comment by SylvainCorlay** [comment]
@jakirkham would you be ok merging this? I would like to make bqplot and pythreejs depend on this.

File: recipes/traittypes/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        - traitlets >=4.2.2
+        - numpy
+
+test:
+    imports:
+        - traittypes
+
+about:
+    home: https://github.com/jupyter-incubator/traittypes
+    license: BSD 3-Clause
```
Comment (jakirkham) [inline recipes/traittypes/meta.yaml]: Could we please add `license_file: LICENSE`?

File: recipes/traittypes/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{% set version = "0.0.4" %}
+{% set md5 = "6378e67cd60c8f8f13bdae2b2a2ef307" %}
```
Comment (jakirkham) [inline recipes/traittypes/meta.yaml]: Just as FYI, [pypi.io]( https://pypi.io ) provides `sha256` checksums.

File: recipes/traittypes/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+about:
+    home: https://github.com/jupyter-incubator/traittypes
+    license: BSD 3-Clause
+    summary: Trait types for NumPy, SciPy and friends
+
+extra:
+  recipe-maintainers:
+    - SylvainCorlay
+    - jasongrout
```
Comment (jakirkham) [inline recipes/traittypes/meta.yaml]: Would be nice if the indentation here matched everything else. Either 2 or 4 spaces should be fine.

**Comment by jakirkham** [comment]
Couple simple things and then would be happy to merge.

**Comment by SylvainCorlay** [comment]
@jakirkham all set.

File: recipes/traittypes/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+test:
+    imports:
+        - traittypes
+
+about:
+    home: https://github.com/jupyter-incubator/traittypes
+    license: BSD 3-Clause
+    summary: Trait types for NumPy, SciPy and friends
+    license_file: LICENSE
```
Comment (jakirkham) [inline recipes/traittypes/meta.yaml]: Would put the license fields together, but can save that for the feedstock.

**Comment by jakirkham** [comment]
Thanks @SylvainCorlay.

**Pull request merged**